controller/api/destination/profile_translator.go

```diff
@@ -23,15 +24,17 @@ type profileTranslator struct {
 	fullyQualifiedName string
 	port               uint32
 	endpoint           *pb.WeightedAddr
+	pod                *corev1.Pod
 }
 
-func newProfileTranslator(stream pb.Destination_GetProfileServer, log *logging.Entry, fqn string, port uint32, endpoint *pb.WeightedAddr) *profileTranslator {
+func newProfileTranslator(stream pb.Destination_GetProfileServer, log *logging.Entry, fqn string, port uint32, endpoint *pb.WeightedAddr, pod *corev1.Pod) *profileTranslator {
```

Comment:
Using this profile_translator for both endpoint profiles (which only contain opaque port info) and for service profiles (which never have an endpoint) might be overloading things.  would it make sense to have `profileTranslator` and a `endpointProfileTranslator` as two separate things?